asciidoctorj-core/src/main/java/org/asciidoctor/internal/JRubyAsciidoctor.java

```diff
@@ -132,8 +132,12 @@ private static void injectEnvironmentVariables(RubyInstanceConfig config, Map<St
 
     private static RubyInstanceConfig createOptimizedConfiguration() {
         RubyInstanceConfig config = new RubyInstanceConfig();
-        config.setCompatVersion(CompatVersion.RUBY2_0);
-        config.setCompileMode(CompileMode.OFF);
+        if (System.getProperty("jruby.compat.version") == null) {
+            config.setCompatVersion(CompatVersion.RUBY2_0);
+        }
+        if (System.getProperty("jruby.compile.mode") == null) {
+            config.setCompileMode(CompileMode.OFF);
+        }
```

Comment:
I think this needs to be the other way around. If the JRuby properties are not specified, we do nothing. I'm not concerned about breaking things because this can be controlled from the outside...and it's really not right that we are setting options without the user's control.